traitlets/traitlets.py

```diff
@@ -423,7 +428,9 @@ def __get__(self, obj, cls=None):
             except KeyError:
                 # Check for a dynamic initializer.
                 if self.name in obj._trait_dyn_inits:
-                    method = getattr(obj, obj._trait_dyn_inits[self.name])
+                    method = obj._trait_dyn_inits[self.name]
+                    if isinstance(method, str):
```

Comment:
N.B. Previously `_trait_dyn_inits` was a mapping of trait name to method name on the object. Now it can contain either a method name or a callable. It's private, and this is the only place we use values from it, I believe.